src/main/webapp/app/exercises/programming/hestia/generation-overview/steps/diff-generation-step/diff-generation-step.component.ts

```diff
@@ -14,20 +14,33 @@ export class DiffGenerationStepComponent implements OnInit {
     exercise: ProgrammingExercise;
 
     @Output()
-    onGitDiffLoaded = new EventEmitter<ProgrammingExerciseFullGitDiffReport>();
+    onGitDiffLoaded = new EventEmitter<ProgrammingExerciseGitDiffReport>();
 
     isLoading = false;
-    gitDiffReport?: ProgrammingExerciseFullGitDiffReport;
+    gitDiffReport?: ProgrammingExerciseGitDiffReport;
+    templateFileContentByPath: Map<string, string>;
+    solutionFileContentByPath: Map<string, string>;
 
     constructor(private exerciseService: ProgrammingExerciseService, private alertService: AlertService) {}
 
     ngOnInit() {
         this.isLoading = true;
-        this.exerciseService.getFullDiffReport(this.exercise.id!).subscribe({
+        this.exerciseService.getDiffReport(this.exercise.id!).subscribe({
             next: (report) => {
                 this.gitDiffReport = report;
-                this.isLoading = false;
                 this.onGitDiffLoaded.emit(report);
+                this.exerciseService.getTemplateRepositoryTestFilesWithContent(this.exercise.id!).subscribe({
+                    next: (response: Map<string, string>) => {
+                        this.templateFileContentByPath = response;
+                        this.isLoading = this.solutionFileContentByPath === undefined;
```

Comment:
Throw an error instead of just a loading spinner? As an end user, I appreciate error messages over unlimited loading animations.